panic: runtime error: invalid memory address or nil pointer dereference

<!--- Please keep this note for the community --->

#### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comment
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

When filing a bug, please include the following headings if possible. Any
example text in this template can be deleted.

#### Overview of the Issue
s/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

#### Overview of the Issue
When i try to run the packer software i get the following error.
```
panic: runtime error: invalid memory address or nil pointer dereference
```

#### Reproduction Steps
I'm trying to run the script by the following command
```bash
packer build ubuntu-server-focal-docker.pkr.hcl
```
(The file is referenced below)

### Packer version
packer version 1.8.4 [go1.18.5 windows amd64]

### Simplified Packer Template
https://raw.githubusercontent.com/MGertz/packer-proxmox/main/ubuntu-server-focal-docker.pkr.hcl

### Operating system and Environment details
WIndows 10, 64bit 22H2

### Log Fragments and crash.log files
https://raw.githubusercontent.com/MGertz/packer-proxmox/main/crash.log

### Repo with the needed files.
https://github.com/MGertz/packer-proxmox

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.com/hashicorp/packer-plugin-proxmox
 go 1.18
 
 require (
-	github.com/Telmate/proxmox-api-go v0.0.0-20230201114211-205f18a553e4
+	github.com/Telmate/proxmox-api-go v0.0.0-20230302112457-ad6090b82842
 	github.com/hashicorp/go-getter/v2 v2.1.1
 	github.com/hashicorp/hcl/v2 v2.14.1
 	github.com/hashicorp/packer-plugin-sdk v0.3.2
@@ -56,6 +56,7 @@ require (
 	github.com/hashicorp/vault/api v1.1.1 // indirect
 	github.com/hashicorp/vault/sdk v0.2.1 // indirect
 	github.com/hashicorp/yamux v0.0.0-20210826001029-26ff87cf9493 // indirect
+	github.com/inconshreveable/mousetrap v1.0.1 // indirect
 	github.com/jehiah/go-strftime v0.0.0-20171201141054-1d33003b3869 // indirect
 	github.com/jmespath/go-jmespath v0.4.0 // indirect
 	github.com/kr/fs v0.1.0 // indirect
@@ -75,6 +76,8 @@ require (
 	github.com/pkg/sftp v1.13.2 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/ryanuber/go-glob v1.0.0 // indirect
+	github.com/spf13/cobra v1.5.0 // indirect
+	github.com/spf13/pflag v1.0.5 // indirect
 	github.com/ugorji/go/codec v1.2.6 // indirect
 	github.com/ulikunitz/xz v0.5.10 // indirect
 	go.opencensus.io v0.23.0 // indirect

